fuzz/db_fuzzer.cc

```diff
@@ -0,0 +1,135 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <filesystem>
+
+#include "rocksdb/db.h"
+
+// Helper function to getting strings based on the random data.
+std::string get_string(const uint8_t** data, size_t* remaining) {
+  if (*remaining == 0) {
+    return std::string("");
+  }
+  uint8_t s_size = (*data)[0];
+
+  // if s_size is more than we have available, do not use s_size
+  if (s_size > *remaining) {
+    std::string s1(reinterpret_cast<const char*>(*data), *remaining);
+    *data += *remaining;
+    *remaining = 0;
+    return s1;
+  }
+
+  std::string s1(reinterpret_cast<const char*>(*data), s_size);
+  *remaining -= s_size;
+  return s1;
+}
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  rocksdb::DB* db;
+  rocksdb::Options options;
+  options.create_if_missing = true;
+  rocksdb::Status status = rocksdb::DB::Open(options, "/tmp/testdb", &db);
```

Comment:
> Thanks for integrating this fuzzer into the `fuzz` module.

No problem, thanks for reviewing! 

> From a high level, what's your thought on using structure-aware fuzzing v.s. interpreting the fuzzer provided `data`?

From a high-level perspective this fuzzer is arguably structure-aware. However, in terms of whether to use libprotobuf-mutator or not, then my personal view is to use it only when there is significant structure on the input to the target, so we can draw benefit of libprotobuf-mutators ability to mutate structures. Examples of what I consider significant structures are things where the data is some form of complex tree-like structure, or where you find need for wrapping the random data in a lot of hard-coded strings.  

In this case, we are essentially just interested in constructing a sequence of commands, with a modest amount of commands available, and that is quite "light" IMO. As such, I personally prefer to not rely libProtobufMutator simply because I can express the "structure" of the sequence with relatively little code in itself.